rosbag2_storage/include/rosbag2_storage/yaml.hpp

```diff
@@ -95,7 +95,8 @@ struct convert<rosbag2_storage::TopicMetadata>
     node["name"] = topic.name;
     node["type"] = topic.type;
     node["serialization_format"] = topic.serialization_format;
-    node["offered_qos_profiles"] = topic.offered_qos_profiles;
+    YAML::Node profiles = YAML::Load(topic.offered_qos_profiles);
+    node["offered_qos_profiles"] = profiles;
```

Comment:
@roncapat Wouldn't you mind to try to pickup this couple of lines changes to better format `offered_qos_profiles` to your  https://github.com/ros2/rosbag2/pull/1476 PR?